### PR TITLE
Allow container upload to multiple registries

### LINKED ISSUE #260: Publish container image somewhere other than docker.io
Due to internal compliance issues, our team is generally unable to pull any containers from docker.io. Almost any other registry would probably work fine; examples of registries we're already pulling from include:

- registry.connect.redhat.com (this is where we currently get Vault from)
- quay.io
- ghcr.io

### PR DESCRIPTION
**Pull request by roceb**
Fixes #260 
Allows pushes to Docker hub, ghcr.io, and quay.io.
Needs secrets to be added for Docker and quay.io.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -90,6 +90,21 @@ jobs:
...
           password: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: "Docker Login: docker.io"
+        if: startsWith(github.ref, 'refs/tags/')
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+
+      - name: "Docker Login: ghcr.io"
```
Comment (cipherboy) [inline .github/workflows/release.yml:100]:
```suggestion
      - name: "Docker Login: quay.io"
```

File: .goreleaser.yaml [hunk 2]
```diff
@@ -438,6 +450,78 @@ docker_manifests:
...
       - ghcr.io/openbao/openbao:{{ .Version }}-ubi-amd64
       - ghcr.io/openbao/openbao:{{ .Version }}-ubi-arm64
+  - name_template: hub.docker.com/openbao/openbao:{{ .Version }}
+    skip_push: false
+    image_templates:
+      - hub.docker.com/openbao/openbao:{{ .Version }}-alpine-amd64
+      - hub.docker.com/openbao/openbao:{{ .Version }}-alpine-arm
+      - hub.docker.com/openbao/openbao:{{ .Version }}-alpine-arm64
+      - hub.docker.com/openbao/openbao:{{ .Version }}-alpine-riscv64
+      - hub.docker.com/openbao/openbao:{{ .Version }}-ubi-amd64
```
Comment (cipherboy) [inline .goreleaser.yaml:460]: @naphelps For upstream, `hashicorp/vault` was always an alpine based image and `hashicorp/vault-ubi` was always the UBI image; do we want to do that downstream as well? That'd be my preference. I recognize this is outside the scope of this PR, but I missed it earlier.

Comment (roceb) [reply .goreleaser.yaml:460]: This does seem outside the scope. Maybe we should open another issue to tackle this.

File: .goreleaser.yaml [hunk 3]
```diff
@@ -228,6 +228,8 @@ dockers:
       - "--target=default"
     image_templates:
       - "ghcr.io/openbao/openbao:{{ .Version }}-alpine-amd64"
+      - "quay.io/openbao/openbao:{{ .Version }}-alpine-amd64"
+      - "hub.docker.com/openbao/openbao:{{ .Version }}-alpine-amd64"
```
Comment (till) [inline .goreleaser.yaml:232]: `hub.docker.com` should be omitted everywhere while building.

Their registry is called `docker.io` but it should work without it.

**Review by till — CHANGES_REQUESTED** [review]
(Sorry for the drive by review)

File: .goreleaser.yaml (hunk 3 above)
Comment (till) [reply .goreleaser.yaml:232]: Also for login, all image templates/manifests in the goreleaser file.

Comment (roceb) [reply .goreleaser.yaml:232]: Also would omitting the name cause confusing in the future?

Comment (till) [reply .goreleaser.yaml:232]: I usually omit it, but whatever you feel like is best. I am also not a maintainer. 🫣

Comment (roceb) [reply .goreleaser.yaml]: @cipherboy what do you recommend? I left it as docker.io for now.

**Review by till — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -181,6 +181,7 @@ jobs:
     strategy:
       matrix:
         arch: [arm, arm64, amd64]
+        registry: [hub.docker.com, quay.io, ghcr.io]
```
[the remark this answers is not in the document]
Comment (roceb) [reply .github/workflows/build.yml]: Should be ready to merge.

**Comment by naphelps** [comment]
@roceb If you could rebase-in updates to your branch from main. I will get this merged in.

**Comment by roceb** [comment]
@naphelps Done

**Pull request merged**